### ISSUE TITLE
Parallelization Issues

### ISSUE THREAD

**Issue by DigitalFlow** [jordimontana82/fake-xrm-easy#74]
Hey Jordi,

just found a bug if requests are executed in parallel:

When executing 10 create requests in a Parallel.ForEach loop, only one record is added to the data dictionary, although FakeItEasy reports that the CreateRequest was executed 10 times on the context.
As I see it this is a problem of using a plain dictionary for storing the data, we should better move to a Concurrent Dictionary.

Kind Regards,
Florian

**Comment by DigitalFlow** [comment]
Hey Jordi,

I'm fine, how about you?

Yes, I meant during a single unit test. So I had one unit test, where records are created in parallel.
Unfortunately after sending ten create requests, only one record made it into our data dictionary.
FakeItEasy however shows, that the CreateRequest was executed multiple times in the Service, so the requests must have made it to the context, but the updates were lost.

**Comment by jordimontana82** [comment]
Good thanks!

Ok, I might have a look.... One thing tho is that I would use ExecuteMultipleRequest for bulk create as opposed to Parallel. Reason is that you would save multiple round trips to the server: 1 single request with the X create messages. 

There is a fake message for that already too.

**Comment by DigitalFlow** [comment]
Right, there is one. I even use ExecuteTransactionRequests in fact.

For this case I wanted to simulate my application however, since it runs on multiple tasks which operate asynchronously.

**Comment by DigitalFlow** [comment]
I think I'll submit a PR which deploys a concurrent dictionary some time soon, I'll let you know if it solved the problem in any case.

**Comment by jordimontana82** [comment]
Cool, let me know and thanks a mill

**Comment by michaelmittermair** [comment]
@jordimontana82 Any update on this since July 2016?

**Comment by jordimontana82** [comment]
@michaelmittermair no updates yet, would you try using a concurrent dictionary like @DigitalFlow suggested above?

**Comment by xairrick** [comment]
I ran into this issue trying to setup test data (product structures are hell).
